src/components/flat-table/flat-table-cell/flat-table-cell.component.tsx

```diff
@@ -59,8 +59,17 @@ export const FlatTableCell = ({
     isFirstCell,
     isExpandableCell,
     makeCellSticky,
+    highlighted,
+    selected,
+    setSelectedId,
   } = useCalculateStickyCells(internalId.current);
 
+  useEffect(() => {
+    if (isExpandableCell && (highlighted || selected)) {
+      setSelectedId?.(internalId.current);
```

Comment:
**comment**: I don't know for sure (it's not trivial to debug as the story you've added for this doesn't precisely replicate the bug scenario), but I suspect this *might* be where the bug I've noted in the overall review comment is coming from. I think if this `if` fails, we need to set the selected ID to something else (ideally to `null` or `undefined` becase surely nothing is selected at that point, although the types don't currently allow that...)

Reply:
Correct we need to unset the id when it's no longer highlighted